icc-x-api/utils/efact-util.ts

```diff
@@ -333,11 +333,16 @@ function toInvoiceItem(
   const invoiceItem = new InvoiceItem({})
   invoiceItem.codeNomenclature = Number(invoicingCode.tarificationId!!.split("|")[1])
   invoiceItem.dateCode = dateEncode(toMoment(invoicingCode.dateCode!!)!!.toDate())
-  if (invoiceItem.codeNomenclature === 109594) {// Prétrajet de soins diabète
-    invoiceItem.endDateCode = dateEncode(
-      toMoment(invoicingCode.dateCode!!)!!
-        .endOf("month")
-        .toDate());
+  // Only applies to flatrate invoicing
+  if(flatrateInvoice) {
+    invoiceItem.endDateCode =
+      invoiceItem.codeNomenclature === 109594 // Diabetes pre-care pathways
+        ? dateEncode(toMoment(invoicingCode.dateCode!!)!!.toDate())
```

Comment:
Are we sure the endDate needs to be specified client-side when code is not 109594? Else safer to leave it untouched and write: 

```
if (flatrateInvoice && invoiceItem.codeNomenclature === 109594) {
  invoiceItem.endDateCode = ....
}
```

Reply:
Indeed yes, this is a special behavior as for flatrate invoicing only.
Diabetes pre-care pathways should be charged with equivalent startDateCode && endDateCode (1st day of month)
While other invoicingCodes should be charged for the entire month (1st to last month's day)